src/core/qgsmapsettings.h

```diff
@@ -864,6 +865,24 @@ class CORE_EXPORT QgsMapSettings : public QgsTemporalRangeObject
      */
     void setCurrentFrame( long long frame );
 
+    /**
+     * Returns the shading renderer used to render shading on the entire map
+     *
+     * \see setShadingRenderer()
+     * \since QGIS 3.30
+     *
+     */
+    const QgsElevationShadingRenderer &shadingRenderer() const;
```

Comment:
Let's rename to elevationShadingRenderer() + setElevationShadingRenderer() for consistency and to make it clear it is related to elevation data...